src/validation/obfl.rng

```diff
@@ -792,24 +792,37 @@
     <element name="table-of-contents">
       <ref name="attlist-table-of-contents"/>
       <oneOrMore>
-        <ref name="toc-entry"/>
+        <ref name="toc-block"/>
       </oneOrMore>
     </element>
   </define>
   <define name="attlist-table-of-contents" combine="interleave">
     <ref name="name-id"/>
   </define>
-  <define name="toc-entry">
-    <element name="toc-entry">
-      <ref name="attlist-toc-entry"/>
+  <define name="toc-block">
+    <element name="toc-block">
+      <ref name="attlist-block"/>
       <zeroOrMore>
         <choice>
-          <ref name="blockContents"/>
           <ref name="toc-entry"/>
+          <ref name="toc-block"/>
+          <ref name="blockContents"/>
+          <ref name="anchor"/>
+          <ref name="block"/>
+          <ref name="table"/>
+          <ref name="xml-data"/>
```

Comment:
As discussed above, ```toc-block``` should not allow text blocks. So blockContents, block, table, and xml-data should not be allowed. Anchors could be discussed but I think they should be a part of entries if they are used.

Reply:
Agree, at least for now everything needs to be in a `toc-entry`. If we allow other things in a `toc-block` it becomes problematic to determine when this content should be rendered and when not. It doesn't make much sense to render these unconditionally.

Anchors could possibly be useful in a `toc-entry`. One could imagine that a toc entry contains a footnote for example. But anyway, let's skip this for now to keep things simple.